Jenkinsfile

```diff
@@ -0,0 +1,55 @@
+/* Only keep the 10 most recent builds. */
+properties([[$class: 'BuildDiscarderProperty',
+                strategy: [$class: 'LogRotator', numToKeepStr: '10']]])
+
+// TODO: Move it to Jenkins Pipeline Library
+
+/* These platforms correspond to labels in ci.jenkins.io, see:
+ *  https://github.com/jenkins-infra/documentation/blob/master/ci.adoc
+ */
+List platforms = ['linux', 'windows']
+Map branches = [:]
+
+for (int i = 0; i < platforms.size(); ++i) {
+    String label = platforms[i]
+    branches[label] = {
+        node(label) {
+            timestamps {
+                stage('Checkout') {
+                    checkout scm
+                }
+
+                stage('Build') {
+                    withEnv([
+                        "JAVA_HOME=${tool 'jdk8'}",
+                        "PATH+MVN=${tool 'mvn'}/bin",
+                        'PATH+JDK=$JAVA_HOME/bin',
+                    ]) {
+                        timeout(30) {
+                            String command = 'mvn --batch-mode clean install -Dmaven.test.failure.ignore=true'
+                            if (isUnix()) {
+                                sh command
+                            }
+                            else {
+                                bat command
+                            }
+                        }
+                    }
+                }
+
+                stage('Archive') {
+                    /* Archive the test results */
+                    junit '**/target/surefire-reports/TEST-*.xml'
+
+                    if (label == 'linux') {
+                      archiveArtifacts artifacts: '**/target/**/*.jar'
+                      findbugs pattern: '**/target/findbugsXml.xml'
+                    }
+                }
+            }
+        }
+    }
+}
+
+/* Execute our platforms in parallel */
+parallel(branches)
```

Comment:
Why not use buildPlugin()?  Is it because this is not a plugin, or is there some other reason that I've missed?

Reply:
Because it's not a plugin. The default flow does not work for libs anymore: there is no HPI file to publish. Even if I workaround that, it will be whack-a-mole. I would rather create a new library module as suggested in TODO

Reply:
@MarkEWaite Also, https://github.com/jenkins-infra/pipeline-library/pull/21 introduces some tooling which makes creating independnent Jenkinsfile methods simple